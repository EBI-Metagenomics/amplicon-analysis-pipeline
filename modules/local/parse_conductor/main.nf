
process PARSE_CONDUCTOR {
    
    // Parse the trimming_conductor output and store 
    // the flags into environment variables 
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(trimming_conductor_out)

    output:
    tuple val(meta), env(fwd_flag), env(rev_flag), emit: conductor_out

    script:
    """
    fwd_flag=\$(sed '1q;d' "${trimming_conductor_out}")
    rev_flag=\$(sed '2q;d' "${trimming_conductor_out}")
    """
}
