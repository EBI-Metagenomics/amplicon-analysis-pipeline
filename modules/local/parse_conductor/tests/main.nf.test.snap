{
    "conductor_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "std",
                    "std"
                ]
            ]
        ],
        "timestamp": "2023-11-03T14:00:19.924220199"
    }
}